python/paddle/nn/quant/format.py

```diff
@@ -0,0 +1,234 @@
+            )
+        else:
+            out = self._helper.create_variable_for_type_inference(input.dtype)
+            self._helper.append_op(
+                type='dequantize_linear',
+                inputs={
+                    'X': input,
+                    'Scale': self._scales,
+                    'ZeroPoint': self._zero_point,
+                },
+                outputs={'Y': out},
+                attrs={
+                    'quant_axis': self._quant_axis,
+                    'bit_length': self._bit_length,
+                },
+            )
+            return out
+
+    @staticmethod
+    def from_quanter(quanter):
+        return LinearDequanter(
+            quanter.scales(),
+            zero_point=quanter.zero_points(),
+            quant_axis=quanter.quant_axis(),
+            bit_length=quanter.bit_length(),
+        )
+
+
+class ConvertibleQuantedLayer(Layer, metaclass=abc.ABCMeta):
+    r"""Abstract class to help convert quantized layer to inference model.
+    It Defined some function convert quanters and observers to quantize or
+    dequantize operators who maitains quantization parameters used during
+    inference.
+    Examples:
+       .. code-block:: python
+
+            # Given codes in ./customized_quanter.py
+            class CustomizedQuantedLayer(ConvertibleQuantedLayer):
+                def __init__(self):
+                    super(CustomizedQuantedLayer, self).__init__()
+                    self.weight_a = paddle.create_parameter(shape=[1], dtype='float32')
+                    self.weight_b = paddle.create_parameter(shape=[1], dtype='float32')
+                    self.quanter_for_weight_a = None
+                    self.activation_weight = None
+                def forward(self, input):
+                    qweight_a = self.quanter_for_weight_a(self.weight_a)
+                    weight_b = self.weight_b
+                    qinput = self.activation_weight(input)
+                    // compute with qweight_a, weight_b and qinput.
+                    return qweight * qinput + weight_b
+
+                def weights_to_quanters(self):
+                    return [('weight_a', 'quanter_for_weight_a')]
+
+                def activation_quanters(self):
+                    return ['activation_weight']
+    """
+
+    def __init__(self):
+        super(ConvertibleQuantedLayer, self).__init__()
+        self.converted = False
+
+    @abc.abstractmethod
+    def weights_to_quanters(self) -> List[Tuple[str, str]]:
+        r"""Get the name pairs of weights to be quantized and corresponding quanters.
+        In convert function of this abstract class, it will call the 'weights_to_quanters' function.
+        And do something as bellow:
+        For each pair, the quanter will be converted to quantize operator and
+        dequantize operator. And then quantize the weight by quantize operator.
+        Finally, remove the quantize operator and store the weights in integer data type.
+
```

Comment:
Get the name pairs of weights to be quantized and their corresponding quantizers. In the convert function of this abstract class, it will call the ‘weights_to_quanters’ function and do something as follows: 
For each pair, the quantizer will be converted to a quantize operator and a dequantize operator. Then, the weight will be quantized by the quantize operator. Finally, the quantize operator will be removed and the weights will be stored in integer data type.